morphia/src/main/java/org/mongodb/morphia/dao/BasicDAO.java

```diff
@@ -24,17 +25,19 @@
  * @author Olafur Gauti Gudmundsson
  * @author Scott Hernandez
  */
-@SuppressWarnings("deprecation")
+@SuppressWarnings({"WeakerAccess", "deprecation", "unused"})
```

Comment:
IDEA suggests that some methods could have weaker access, e.g., `protected` -> `private`.  Because this change would be unrelated and possibly breaking in some cases, this suppresses that suggestion.